src/lib/core/OTAImageHeader.h

```diff
@@ -58,6 +58,62 @@ struct OTAImageHeader
     ByteSpan mImageDigest;
 };
 
-CHIP_ERROR DecodeOTAImageHeader(ByteSpan buffer, OTAImageHeader & header);
+class OTAImageHeaderParser
+{
+public:
+    /**
+     * @brief Prepare the parser for accepting Matter OTA image chunks.
+     *
+     * The method can be called many times to reset the parser state.
+     */
+    void Init();
+
+    /**
+     * @brief Clear all resources associated with the parser.
+     */
+    void Clear();
+
+    /**
+     * @brief Returns if the parser is ready to accept subsequent Matter OTA image chunks.
+     */
+    bool IsInitialized() const { return mState != State::kNotInitialized; }
+
+    /**
+     * @brief Decode Matter OTA image header
```

Comment:
I think it would be very helpful to have more details here on how this API is meant to be used for parsing a header that comes in multiple chunks. For example, it looks like any return other than CHIP_NO_ERROR indicates an error. So is it a legitimate situation to have the buffer contain only a part of the header? If so, what will be the return value?

Reply:
I thought I described this in the section about error codes, but I repeated that information to make it more clear.